p2p/transport/webrtc/udp_mux.go

```diff
@@ -0,0 +1,351 @@
+package libp2pwebrtc
+
+// udpMuxNewAddr is mostly similar to UDPMuxDefault exported from
+// pion/ice [https://github.com/pion/ice/blob/master/udp_mux.go] with
+// the only difference being the additional channel to notify the libp2p
+// transport of any STUN requests from unknown ufrags.
+
+import (
+	"errors"
+	"io"
+	"net"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/pion/ice/v2"
+	"github.com/pion/logging"
+	"github.com/pion/stun"
+)
+
+type candidateAddr struct {
+	raddr *net.UDPAddr
+	ufrag string
+}
+
+var _ ice.UDPMux = &udpMuxNewAddr{}
+
+// udpMuxNewAddr is an implementation of the interface
+type udpMuxNewAddr struct {
+	params ice.UDPMuxParams
+
+	closedChan chan struct{}
+	closeOnce  sync.Once
+
+	// connsIPv4 and connsIPv6 are maps of all udpMuxedConn indexed by ufrag|network|candidateType
+	connsIPv4, connsIPv6 map[string]*udpMuxedConn
+
+	addressMapMu sync.RWMutex
+	addressMap   map[string]*udpMuxedConn
```

Comment:
I'm not sure I understand the logic here. Why are the conns maps defined above not sufficient? It would be nice to just have a single `map[string]*udpMuxedConn`.
If you think this map is really need, can you please add a comment explaining the purpose and how it differs from the conn map?

Reply:
This has been completely refactored.